src/wasmcloud.rs

```diff
@@ -175,3 +182,35 @@ fn extract_arg_value(arg: &str) -> Result<String> {
         Err(_) => Ok(arg.to_string()),
     }
 }
+
+/// Parse a single key-value pair
+fn parse_key_val<T, U>(s: &str) -> std::result::Result<(T, U), Box<dyn Error>>
+where
+    T: str::FromStr,
+    T::Err: Error + 'static,
+    U: str::FromStr,
+    U::Err: Error + 'static,
+{
+    let pos = s
+        .find('=')
+        .ok_or_else(|| format!("invalid KEY=value: no `=` found in `{}`", s))?;
+    Ok((s[..pos].parse()?, s[pos + 1..].parse()?))
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn parse_key_val_ok() {
+        let expected = ("x".to_string(), "1".to_string());
+        let actual = parse_key_val::<String, String>("x=1").unwrap();
+        assert!(actual == expected);
+    }
+
+    #[test]
+    fn parse_key_val_err() {
+        let actual = parse_key_val::<String, String>("bad string");
+        assert!(actual.is_err());
+    }
```

Comment:
Thank you for adding tests! This is awesome 😄

Reply:
pleasure :-) I'm just going to tweak that first one, to make it a bit more idiomatic.

Reply:
just removed the turbofish, and changed the assert. Reads better, and a failure would be better reported